File: frontend/patch_quiz_double.js
const fs = require('fs');
const path = require('path');
const file = 'c:\\Users\\nminh\\Desktop\\LearnReact\\frontend\\src\\features\\homework\\pages\\MyHomeworkLessonPage.jsx';
let content = fs.readFileSync(file, 'utf8');

// 1. Remove the hidden sm:block from the main quiz body so it shows everywhere
content = content.replace(
    '<div className="homework-quiz-body !p-0 hidden sm:block">',
    '<div className="homework-quiz-body !p-0">'
);

// 2. Remove the duplicated secondary block entirely
const startMatch = '<div className="homework-quiz-body block sm:hidden">';
const startIdx = content.indexOf(startMatch);

if (startIdx !== -1) {
    // Find where it ends
    const endMatch = '      <p className="homework-item-meta">Selections are kept locally on this page while you work.</p>';
    const endIdx = content.indexOf(endMatch, startIdx);

    if (endIdx !== -1) {
        // Cut out the duplicate list entirely
        content = content.substring(0, startIdx) + content.substring(endIdx);
    }
}

fs.writeFileSync(file, content, 'utf8');
console.log('Fixed double rendering in MyHomeworkLessonPage.jsx');

File: frontend/patch_quiz_double.py
import os

file_path = "src/features/homework/pages/MyHomeworkLessonPage.jsx"
with open(file_path, "r", encoding="utf-8") as f:
    content = f.read()

# 1. Remove hidden sm:block
content = content.replace(
  '<div className="homework-quiz-body !p-0 hidden sm:block">',
  '<div className="homework-quiz-body !p-0">'
)

# 2. Remove the duplicated secondary block explicitly
start_match = '<div className="homework-quiz-body block sm:hidden">'
end_match = '      <p className="homework-item-meta">Selections are kept locally on this page while you work.</p>'

start_idx = content.find(start_match)
if start_idx != -1:
    end_idx = content.find(end_match, start_idx)
    if end_idx != -1:
        content = content[:start_idx] + content[end_idx:]

with open(file_path, "w", encoding="utf-8") as f:
    f.write(content)

print("Fixed double rendering in MyHomeworkLessonPage.jsx")
